appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/UberJarCreator.java

```diff
@@ -241,6 +251,19 @@ public void buildUberJar() {
                     }
                 }
             }
+            
+            if (copyDirectory != null) {
+                String basePath = copyDirectory.getPath().replaceAll(copyDirectory + "$", "");
+                List<File> filesToCopy = fillFiles(copyDirectory);
+                for (File file : filesToCopy) {
+                    
+                        JarEntry deploymentEntry = new JarEntry(copyDirectory + file.getPath().replace(basePath, ""));
+                        jos.putNextEntry(deploymentEntry);
+                        Files.copy(file, jos);
```

Comment:
I think this would only be necessary if the JARs are to be placed into the MICRO-INF/lib folder. But it wouldn't hurt to convert all JARs to uncompressed JARs, because it would be very rare that users would want to place JARs somewhere else.